plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeHistoryTable.java

```diff
@@ -41,26 +50,34 @@
 import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
 import static io.trino.spi.type.TypeSignature.mapType;
 import static io.trino.spi.type.VarcharType.VARCHAR;
-import static java.util.Comparator.comparingLong;
 import static java.util.Objects.requireNonNull;
 
 public class DeltaLakeHistoryTable
         implements SystemTable
 {
+    private final SchemaTableName tableName;
     private final ConnectorTableMetadata tableMetadata;
     private final List<Type> types;
-    private final Iterable<CommitInfoEntry> commitInfoEntries;
+    private final TrinoFileSystemFactory fileSystemFactory;
+    private final DeltaLakeMetastore metastore;
+
     private final Type varcharToVarcharMapType;
 
-    public DeltaLakeHistoryTable(SchemaTableName tableName, List<CommitInfoEntry> commitInfoEntries, TypeManager typeManager)
+    public DeltaLakeHistoryTable(
+            SchemaTableName systemTableName,
```

Comment:
I'd rather just take the main table name as an argument and append the `$history` part back in here.